### PR TITLE
Linking in assembly mode

### PR DESCRIPTION
**Pull request by cameel**
**This PR is based on #10227. Please don't merge until that one is merged.**

Note: There's no linking when `--machine wasm` is used. I could have added it but I would not be able to check if it works because Yul code with `linkersymbol()` won't get translated to wasm yet anyway (#10173).

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ Compiler Features:
  * SMTChecker: Support ``selector`` for expressions with value known at compile-time.
  * Command Line Interface: New option ``--model-checker-timeout`` sets a timeout in milliseconds for each individual query performed by the SMTChecker.
  * Standard JSON: New option ``modelCheckerSettings.timeout`` sets a timeout in milliseconds for each individual query performed by the SMTChecker.
+ * Commandline Interface: Perform linking in assembly mode when library addresses are provided.
+ * Standard JSON: Perform linking in assembly mode when library addresses are provided.
```
Comment (cameel) [inline Changelog.md]: Should I be adding two entries when the change affects both CLI and Standard JSON?

**Comment by axic** [comment]
Just mark it draft and can't be accidentally merged.

**Comment by cameel** [comment]
But this is ready for review. It's just that it needs to be merged in the right order. Merging it into `develop` would also merge the PR below.

File: Changelog.md (hunk 1 above)
Comment (chriseth) [reply Changelog.md]: What about `Assembler: ...`? It's not really a change to the commandline interface

**Review by chriseth — COMMENTED** [review]
Looks good!

File: Changelog.md (hunk 1 above)
Comment (cameel) [reply Changelog.md]: You have a point :) `Assembler` sounds good.

Comment (cameel) [reply Changelog.md]: Done.

**Review by chriseth — APPROVED** [review]

**Pull request merged**